FreeRTOS/FreeRTOS-Plus-TCP
# Update IPv6 branch according to main

Description
-----------
This change takes care of resolving all conflicts between dev/IPv6_integration and main branch.
NetworkInterface/NXP1060/NetworkInterface changes will be taken care in a separate PR.

Test Steps
-----------
Jenkins run - https://amazon-freertos-ci.corp.amazon.com/job/freertos_distro/job/custom_job_pipeline/990/

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
